src/Perspex.Controls/Primitives/Popup.cs

```diff
@@ -273,9 +273,30 @@ private Point GetPosition()
                     case PlacementMode.Bottom:
                         point = new Point(0, target.Bounds.Height);
                         break;
+
                     case PlacementMode.Right:
                         point = new Point(target.Bounds.Width, 0);
                         break;
+
+                    case PlacementMode.Center:
+
+                        Size childSize;
+
+                        if ((Child != null) && (_topLevel != null))
+                        {
+                            Child.Measure(_topLevel.ClientSize);
```

Comment:
It works only the first time the popup is shown. After that, after calling PopupRoot.LayoutManager.ExecuteLayoutPass(), PopupRoot.DesizedSize is (0,0).

Never mind, I changed it to calculate according to target as for the other  placement modes.